cmd/performance-profile-creator/cmd/root.go

```diff
@@ -32,6 +33,13 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// ProfileData collects and stores all the data needed for profile creation
+type ProfileData struct {
+	isolatedCPUs, reservedCPUs string
+}
+
+var profileData ProfileData
```

Comment:
why global? can't be moved in the `PreRunE` body?

Reply:
I would really like to do this in a bit better and am open to suggestions here but reason for the current implementation is becase handle.GetReservedAndIsolatedCPUs is called in PreRunE where we populate the profileData and we need this data in the Run() where call to createProfile(profileName, **profileData**)

Reply:
OK, silly questions first: can't we merge `PreRunE` and `Run` in just one `RunE`? I get why we need the -E variant, I'm still not sure why we need PreRun + Run.

Reply:
It looks much better since it returns error making it the perfect place to validate the input.
We could (probably) use only Run, but then why use cobra at all if we don't use their features?

Also while I am against global variables, every piece of software has a "context" that can be passed between different subsystems.  If cobra supports such context, then fine, we should use the cobra context to pass our data between the calls. Otherwise using a global context is also acceptable for a small piece of software as long as the "global context" is private to the package/file and not accessible from all the code.  

On the other hand, if Cobra can't help us to complete our task, it becomes an obstacle rather than help us. I am not against Cobra, it seems cool.

Reply:
But it seems we may not need the context after all, see bellow

Reply:
uhm, good point.

Reply:
Okay, what I am going to do is remove Run and move the profile creation to RunE instead.